src/bio.c

```diff
@@ -670,7 +670,7 @@ int wolfSSL_BIO_write(WOLFSSL_BIO* bio, const void* data, int len)
     if (front != NULL && front->infoCb != NULL) {
         ret = (int)front->infoCb(front,
                                  WOLFSSL_BIO_CB_WRITE | WOLFSSL_BIO_CB_RETURN,
-                                 (const char*)data, 0, 0, ret);
+                                 (const char*)data, len, 0, ret);
```

Comment:
In the first call to `infoCb` in this function, the operation is `WOLFSSL_BIO_CB_WRITE`. libimobiledevice's write callback will only write if the operation is `WOLFSSL_BIO_CB_WRITE | WOLFSSL_BIO_CB_RETURN`. The result is that the first `infoCb` call just returns immediately. This second one does the writing, but, it can only do that if it knows the length.